app/components/pagination-group.js

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class PaginationGroupComponent extends Component {
+    get pageValues() {
+        const pages = Array.from({ length: Math.ceil(this.args.model.meta.count / this.args.limit) }, (_, i) => i + 1);
+
+        return pages;
+    }
+
+    limitValues = [5, 10, 20, 50];
+
+    resetPage(newLimit) {
+        newPage = 1;
+
+        this.args.updatePaginationGroup(newPage, newLimit);
+    }
+
+    movePage(amount) {
+        newPage = this.args.page + amount;
+        newLimit = this.args.limit;
+
+        this.args.updatePaginationGroup(newPage, newLimit);
+    }
+
+    @action
+    onPageClick(newPage) {
+        newLimit = this.args.limit;
+
+        this.args.updatePaginationGroup(newPage, newLimit);
```

Comment:
Looks like this can be moved to some function like updatePagination

Reply:
Done